File: Java_Assignment/Assignment_58/programAQ1.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement58Question1.java
//  Description : java program to print pyramid of characters using class 
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*

Enter Number : 5
        A
      A B A
     A B C B A 
   A B C D C B A
  A B C D E D C B A

*/

import java.util.Scanner;

class Pattern2
{
    private int num;
    
    public void setNum(int num)
    {
        this.num=num;
    }
    
    public int getNum()
    {
        return this.num;
    }
    
    public void printAnswer()
    {
        int no = getNum();

        for(int i = 0;i < no;i++)
        {
            for(int k = 0;k < no - i - 1;k++)
            {
                System.out.print(" ");
            }
            
            for(int j = 0;j <= i;j++)
            {
                System.out.print((char)('A' + j) + " ");
            }

            for(int j = i -1;j >= 0;j--)
            {
                System.out.print((char)('A' + j) + " ");
            }
            System.out.println();
        }

    }
    
    public void inputNum()
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter Number : ");
        int num = sc.nextInt();
        setNum(num);
        sc.close();
    }
    public static void main(String[] ar)
    {
        Pattern2 ob = new Pattern2();
        ob.inputNum();
        ob.printAnswer();
    }


}

File: C_Assignment/Assignment_29/programAQ1.c
///////////////////////////////////////////////////////////
//
//  File name : Assignement29Question1.c
//  Description : Accepts file name from user and counts number 
//                of capital characters from that file.
//  Author : Kartik Ganesh Jare
//  Date : 13/6/25
//  
///////////////////////////////////////////////////////////
/*

Input : Demo.txt

Output : Number of capital characters are 23

*/

#include <stdio.h>
#include <fcntl.h>
#include <string.h>

#define BUFFER_SIZE 1024

int CountCapital(char FName[])
{
    int fd = 0, iRet = 0, iCnt = 0, iCapCount = 0;
    char Buffer[BUFFER_SIZE] = {'\0'};

    fd = open(FName, O_RDONLY);
    if(fd == -1)
    {
        return -1;
    }

    while((iRet = read(fd, Buffer, BUFFER_SIZE)) > 0)
    {
        for(iCnt = 0; iCnt < iRet; iCnt++)
        {
            if((Buffer[iCnt] >= 'A') && (Buffer[iCnt] <= 'Z'))
            {
                iCapCount++;
            }
        }
    }

    close(fd);
    return iCapCount;
}

int main()
{
    char FileName[60] = {'\0'};
    int iRet = 0;

    printf("Please enter the file name : ");
    scanf("%[^\n]s", FileName);

    iRet = CountCapital(FileName);
    if(iRet == -1)
    {
        printf("Unable to open file");
        return -1;
    }
    else
    {
        printf("Number of Capital characters are %d\n", iRet);
    }
    
    return 0;
}